orb_models/dataset/ase_dataset.py

```diff
@@ -0,0 +1,258 @@
+        edge_properties = _get_properties(property_config.edge_properties)
+        system_properties = _get_properties(property_config.graph_properties)
+        return {
+            "node": node_properties,
+            "edge": edge_properties,
+            "graph": system_properties,
+        }
+
+    def _add_extra_feats_and_targets(
+        self,
+        atom_graph: AtomGraphs,
+        extra_feats: Dict[str, Dict],
+        extra_targets: Dict[str, Dict],
+    ):
+        """Add extra features and targets to the AtomGraphs object.
+
+        Args:
+            atom_graph: AtomGraphs object to add extra features and targets to.
+            extra_feats: Dictionary of extra features with keys
+            extra_targets: Dictionary of extra targets to add.
+        """
+        node_feats = {**atom_graph.node_features, **extra_feats["node"]}
+        edge_feats = {**atom_graph.edge_features, **extra_feats["edge"]}
+
+        system_feats = (
+            atom_graph.system_features if atom_graph.system_features is not None else {}
+        )
+        system_feats = {**system_feats, **extra_feats["graph"]}
+
+        node_targets = (
+            atom_graph.node_targets if atom_graph.node_targets is not None else {}
+        )
+        node_targets = {**node_targets, **extra_targets["node"]}
+
+        edge_targets = (
+            atom_graph.edge_targets if atom_graph.edge_targets is not None else {}
+        )
+        edge_targets = {**edge_targets, **extra_targets["edge"]}
+
+        system_targets = (
+            atom_graph.system_targets if atom_graph.system_targets is not None else {}
+        )
+        system_targets = {**system_targets, **extra_targets["graph"]}
+
+        return atom_graph._replace(
+            node_features=node_feats,
+            edge_features=edge_feats,
+            system_features=system_feats,
+            node_targets=node_targets if node_targets != {} else None,
+            edge_targets=edge_targets if edge_targets != {} else None,
+            system_targets=system_targets if system_targets != {} else None,
+        )
+
+
+def get_dataset(
+    path: Union[str, Path],
+    name: str,
+    system_config: atomic_system.SystemConfig,
+    target_config: atomic_system.PropertyConfig,
+) -> AseSqliteDataset:
+    """Dataset factory function."""
+    return AseSqliteDataset(
+        path=path,
+        name=name,
+        system_config=system_config,
+        target_config=target_config,
+    )
+
+
+def random_rotations_with_properties(
+    atoms: ase.Atoms, properties: dict
+) -> Tuple[ase.Atoms, dict]:
+    """Randomly rotate atoms in ase.Atoms object.
+
+    This exists to handle the case where we also need to rotate properties.
+    Currently we only ever do this for random rotations, but it could be extended.
+
+    Args:
+        atoms (ase.Atoms): Atoms object to rotate.
+        properties (dict): Dictionary of properties to rotate.
+    """
+    rand_rotation = o3.rand_matrix(1)[0].numpy()
```

Comment:
This introduces a dependency on e3nn, I would like to avoid this - can we make our own rotations

Reply:
I've removed the e3nn dependency. The original matrix rotation is actually a bit convoluted, so we end up with more functions in the `utils.py` file.